forest/src/daemon.rs

```diff
@@ -151,10 +155,22 @@ pub(super) async fn start(config: Config) {
     .unwrap();
     chain_store.set_genesis(&genesis.blocks()[0]).unwrap();
 
+    // Reward calculation is needed by the VM to calculate state, which can happen essentially anywhere the `StateManager` is called.
```

Comment:
I like it. We can't statically type it but I think using dynamic dispatching for this is an elegant solution.

Reply:
Cool, yes it's not invoked often, it shouldn't matter. The Forest devs reckon the whole consensus should be config based.